### DISCUSSION TITLE
widget scaling

### DISCUSSION THREAD

**Discussion by Solidity-X** [TomSchimansky/CustomTkinter#1672] [Q&A]
Hey guys!
So I have an application that I want to scale based on the window height/width
first of all I listened to any events with:
`tracker = root.bind("<Configure>", resize)`
Then I just filtered for resize events by checking if the event.widget is root and if the last saved width/height changed to the current.
`if event.widget == root and (width != event.width or height != event.height):`
Now I can use `ctk.set_widget_scaling(ratio)` to set the scaling ratio and scale everything up based on the ratio provided.
I have no problem getting the ratio and everything. But for some reason `ctk.set_widget_scaling(ratio)` is very slow and cancels the whole resizing which means that if I resize the window and drag the coursor then it'll break and I need to click on the window again and drag it so it kinda canceln the whole thing.
Does anyone have alternatives for this resize or scaling or any way to speed it up and not let it cancle the whole thing?

**Comment by avalon60** [comment]
I can't say that I am surprised. Depending on the app's complexity, the widget resizing appears to do some heavy lifting.

I think the answer lies in [this thread](https://stackoverflow.com/questions/40284580/tkinter-get-buttonrelease-1-when-resizing-window) (see Bryan Oakley's response). 

I guess the bit that might be missing is the resetting of the `self._after_id` to `None`, which I think would need to be done in `self.redraw` method. Anyway, this technique essentially offsets any re-renderihg, by using `after`, to prevent a rendering storm.
